public/app/features/dashboard-scene/settings/VariablesEditView.tsx

```diff
@@ -190,6 +201,7 @@ function VariableEditorSettingsListView({ model }: SceneComponentProps<Variables
           navModel={navModel}
           dashboard={dashboard}
           onDelete={onDelete}
+          isVariableNameTaken={isVariableNameTaken}
```

Comment:
What do you think about passing here a function `onValidateName`, which validates the variable and returns `[boolean, errorMessage]`? 

I'm not very fan of this API, too specific 😓  

```suggestion
          validateVariableName={() => model.validateVariableName(variable.name)}
```

This is just an idea, maybe we can find something better 🙇